src/sdk/namespace_fs.js

```diff
@@ -482,6 +609,7 @@ class NamespaceFS {
                     }
                     for (let i = marker_index; i < sorted_entries.length; ++i) {
                         const ent = sorted_entries[i];
+                        if (list_versions && !ent.name.includes(marker_curr)) break;
```

Comment:
can you explain this condition? what happens if marker_curr is 'c' but object name is 'ac'?

Reply:
agree. It should be `match` not `includes`.

When the `key_marker` is provided then we only have to list the versions of that marker. So breaking here if the key is other than `key_marker`